data-prepper-plugins/sns-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/sns/SnsSinkServiceTest.java

```diff
@@ -0,0 +1,155 @@
+            "        message_group_id: test\n" +
+            "        aws:\n" +
+            "          region: ap-south-1\n" +
+            "          sts_role_arn: arn:aws:iam::524239988912:role/app-test\n" +
+            "          sts_header_overrides: {\"test\":\"test\"}\n" +
+            "        codec:\n" +
+            "          ndjson:\n" +
+            "        max_retries: 10\n" +
+            "        dlq_file: C:\\Work\\dlq.txt\n";
+    private PublishBatchResponse publishBatchResponse;
+
+    private SdkHttpResponse sdkHttpResponse;
+
+    @BeforeEach
+    public void setup() throws JsonProcessingException {
+        this.snsSinkConfig = objectMapper.readValue(config, SnsSinkConfig.class);
+        this.snsClient = mock(SnsClient.class);
+        this.pluginMetrics = mock(PluginMetrics.class);
+        this.numberOfRecordsSuccessCounter = mock(Counter.class);
+        this.numberOfRecordsFailedCounter = mock(Counter.class);
+        this.publishBatchResponse = mock(PublishBatchResponse.class);
+        this.sdkHttpResponse = mock(SdkHttpResponse.class);
+        when(pluginMetrics.counter(SnsSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_SUCCESS)).thenReturn(numberOfRecordsSuccessCounter);
+        when(pluginMetrics.counter(SnsSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED)).thenReturn(numberOfRecordsFailedCounter);
+        when(snsClient.publishBatch(any(PublishBatchRequest.class))).thenReturn(publishBatchResponse);
+        when(publishBatchResponse.sdkHttpResponse()).thenReturn(sdkHttpResponse);
+        when(publishBatchResponse.sdkHttpResponse().statusCode()).thenReturn(new Random().nextInt());
+        when(sdkHttpResponse.statusCode()).thenReturn(200);
+    }
+
+
+    private SnsSinkService createObjectUnderTest(){
+        return new SnsSinkService(snsSinkConfig,
+                snsClient,
+                pluginMetrics, mock(PluginFactory.class),
+                mock(PluginSetting.class),
+                mock(ExpressionEvaluator.class));
+    }
+
+    @Test
+    public void sns_sink_test_with_empty_collection_records(){
+        numberOfRecordsSuccessCounter = mock(Counter.class);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        snsSinkService.output(List.of());
+        verifyNoInteractions(snsClient);
+        verifyNoInteractions(numberOfRecordsSuccessCounter);
+        verifyNoInteractions(numberOfRecordsFailedCounter);
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_success_push_to_sns(){
+        when(sdkHttpResponse.statusCode()).thenReturn(500);
+        SnsSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(numberOfRecordsSuccessCounter).increment(records.size());
+    }
```

Comment:
I mentioned this in an earlier comment.
https://github.com/opensearch-project/data-prepper/pull/2995#discussion_r1277804874

We need to verify a couple of things:

1. The sink calls the client.
2. The data that the sink provides to the client is correct.

It should look somewhat like the following.

final PublishBatchRequest request = createPublicRequestByTopic(topicName, processRecordsList);
            final PublishBatchResponse publishBatchResponse = snsClient.publishBatch(request);

```
final ArgumentCaptor<PublishBatchRequest> publishBatchRequestCaptor = ArgumentCaptor.forClass(PublishBatchRequest.class);
verify(snsClient).publishBash(publishBatchRequestCaptor.capture());

final PublishBatchRequest actualRequest = publishBatchRequestCaptor.getValue();

assertThat(actualRequest.topicArn(), equalTo(topicArn));
assertThat(actualRequest.publishBatchRequestEntries().size(), equalTo(1));
// Dive into the first publish batch request entry and verify that it has the correct string.
```

Reply:
Thanks @dlvenable for your feedback, we have incorporated the changes and available for review.